server.js

```diff
@@ -1,23 +1,110 @@
 const express = require("express");
-const cors = require("cors");
+const cors = require('cors')
+var bodyParser = require('body-parser');
 
 const app = express();
-
-app.use(cors());
-
+const port = 3001;
+app.use(cors())
+app.use(bodyParser.urlencoded({ extended: true })); 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
-};
+  timeSent: new Date(2020, 11, 17, 3, 24, 0)
+}
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const messages = [welcomeMessage]
+
+
+app.get('/', function(request, response) {
+  response.sendFile(__dirname + '/index.html');
+});
+//reading all messages
+app.get('/messages', function(request, response) {
+  response.json(messages);
+});
+
+//creating new message
+app.post('/messages',  (req, res) => {
+  let message={
+    id:messages.length,
+    from:req.body.from ,
+    text:req.body.text,
+    timeSent:new Date()
+  }
+  if ((message.from.length>0)&&(message.text.length>0))
+  {
+messages.push(message);
+  res.json(messages);
+}
+  else
+    res.status(400).send(" please enter a value in Name and Message")
+});
+
+
+//Reading one message specified by an 
+app.get('/messages/:messageId',  (req, res) => {
+  const messageId= req.params.messageId;
+  const message = messages.find(p => p.id === parseInt(messageId));
+  if(message){
+  res.json(message);
+  
+
+  }
+  else
+    res.status(400).send(" wrong request")
+});
+// deleting a message
+app.delete('/messages/:message_id',  (req, res) => {
+  const messageId = Number(req.params.message_id);
+  let requiredIndex=messages.findIndex(m=>m.id===messageId);
+  if(requiredIndex){
+  messages.splice(requiredIndex,1)
+  res.json(messages);
+  }
+  else {
+    res.status(400).send(" wrong request")
+  }
+
+})
```

Comment:
There is a problem with your use of findIndex and splice. Try doing a deletion with an ID number that doesn't exist, for example with an array of 3 messages with IDs 0,1,2, try deleting ID 10. It will delete the last message in the array. 
If you add some debug lines, using console.log() you can see what is returned when findIndex does not find the ID, and work out the cause of the problem.

Reply:
Nice point! I think the error case wasn't tested properly here because then you must have seen it doesn't work.
Please pay attention to check all the possible outcomes of your code.
Another aspect of the findIndex method here. Let's say that you try to delete a message with 0 index. What would be then the outcome of this `if(requiredIndex)`?

For more help read the documentation of findIndex and splice.